out_writeapi_test.go

```diff
@@ -38,3 +45,223 @@ func TestFLBPluginRegister(t *testing.T) {
+			checks.appendRows++
+
+			var combinedData []byte
+			for _, tempData := range data {
+				combinedData = append(combinedData, tempData...)
+			}
+
+			message := dynamicpb.NewMessage(md)
+			err := proto.Unmarshal(combinedData, message)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to unmarshal")
+			}
+
+			textField := message.Get(md.Fields().ByJSONName("Text"))
+			timeField := message.Get(md.Fields().ByJSONName("Time"))
+
+			assert.Equal(t, "FOO", textField.String())
+			assert.Equal(t, "000", timeField.String())
+
+			return nil, nil
+		},
+		FinalizeFunc: func(ctx context.Context, opts ...gax.CallOption) (int64, error) {
+			return 0, nil
+		},
+		FlushRowsFunc: func(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error) {
+			return 0, nil
+		},
+		StreamNameFunc: func() string {
+			return ""
+		},
+	}
+
+	origFunc := getWriter
+	getWriter = func(client ManagedWriterClient, ctx context.Context, projectID string, opts ...managedwriter.WriterOption) (MWManagedStream, error) {
+		return mockMS, nil
+	}
+	defer func() { getWriter = origFunc }()
+
+	patch1 := monkey.Patch(output.FLBPluginConfigKey, func(plugin unsafe.Pointer, key string) string {
+		return ""
+	})
+	defer patch1.Unpatch()
+
+	patchSetContext := monkey.Patch(output.FLBPluginSetContext, func(plugin unsafe.Pointer, ctx interface{}) {
+		setID = ctx.(int)
+	})
+	defer patchSetContext.Unpatch()
+
+	plugin := unsafe.Pointer(nil)
+	initRes := FLBPluginInit(plugin)
+
+	orgFunc := getFLBPluginContext
+	getFLBPluginContext = func(ctx unsafe.Pointer) int {
+		checks.calledGetContext++
+		return setID
+	}
+	defer func() { getFLBPluginContext = orgFunc }()
+
+	origReadyFunc := isReady
+	isReady = func(queueHead *managedwriter.AppendResult) bool {
+		// Response is always ready for test
+		return true
+	}
+	defer func() { isReady = origReadyFunc }()
+
+	origResultFunc := pluginGetResult
+	pluginGetResult = func(queueHead *managedwriter.AppendResult, ctx context.Context) (int64, error) {
+		checks.getResultsCount++
+		return -1, nil
+	}
+	defer func() { pluginGetResult = origResultFunc }()
+
+	patchDecoder := monkey.Patch(output.NewDecoder, func(data unsafe.Pointer, length int) *output.FLBDecoder {
+		checks.createDecoder++
+		return nil
+	})
+	defer patchDecoder.Unpatch()
+
+	var rowSent int = 0
+	var rowCount int = 5
+	patchRecord := monkey.Patch(output.GetRecord, func(dec *output.FLBDecoder) (ret int, ts interface{}, rec map[interface{}]interface{}) {
+		checks.gotRecord++
+		dummyRecord := make(map[interface{}]interface{})
+		if rowSent < rowCount {
+			rowSent++
+			// Represents "FOO" in bytes as the data for the Text field
+			dummyRecord["Text"] = []byte{70, 79, 79}
+			// Represents "000" in bytes as the data for the Time field
+			dummyRecord["Time"] = []byte{48, 48, 48}
+			return 0, nil, dummyRecord
+		}
+		rowSent = 0
```

Comment:
Perhaps add a comment that this is being reset in preparation for the next series of invocations of output.GetRecord.

Reply:
Done.